src/main/java/uk/gov/indexer/dao/IndexedEntriesUpdateDAO.java

```diff
@@ -7,10 +7,17 @@
 
 public interface IndexedEntriesUpdateDAO extends DBConnectionDAO {
     String INDEXED_ENTRIES_TABLE = "ORDERED_ENTRY_INDEX";
+    String INDEXED_ENTRIES_INDEX = INDEXED_ENTRIES_TABLE + "_GIN";
 
     @SqlUpdate("CREATE TABLE IF NOT EXISTS " + INDEXED_ENTRIES_TABLE + " (SERIAL_NUMBER INTEGER PRIMARY KEY, ENTRY JSONB)")
     void ensureIndexedEntriesTableExists();
 
+    @SqlUpdate("CREATE INDEX " + INDEXED_ENTRIES_INDEX + " ON " + INDEXED_ENTRIES_TABLE + " USING gin(ENTRY jsonb_path_ops)")
```

Comment:
once postgres 9.5x available we can use query like "CREATE INDEX IF NOT EXISTS".